lib/common/busypoll.c

```diff
@@ -0,0 +1,448 @@
+#include "h2o.h"
+
+#include "h2o/busypoll.h"
+#include <linux/genetlink.h>
+#include <linux/netlink.h>
+#include <linux/filter.h>
+#include <linux/if.h>
+#include <sys/ioctl.h>
+#include <sys/sysinfo.h>
+#include <ynl/ynl.h>
+#include <ynl/netdev-user.h>
+
+static __thread int cpu_claimed = 0;
+static __thread const char *nic_claimed = NULL;
+static __thread unsigned int napi_id_claimed = 0;
+
+static void setup_queue(uint32_t cfg_ifindex, uint32_t cfg_defer_hard_irqs, uint64_t cfg_gro_flush_timeout,
+                        uint64_t cfg_irq_suspend_timeout, uint32_t *napi_ids, size_t napi_ids_size)
+{
+    struct netdev_queue_get_req *get_req = NULL;
+    struct netdev_napi_set_req *set_req = NULL;
+    struct ynl_sock *ys;
+    struct ynl_error yerr;
+
+    ys = ynl_sock_create(&ynl_netdev_family, &yerr);
+    if (!ys)
+        h2o_fatal("ynl, failed to create socket: %s\n", yerr.msg);
+
+    if (cfg_ifindex == 1)
```

Comment:
i dont really follow this -- is this assuming that the first ifindex is always going to be `lo` and thus no NAPI IDs can be fetched? 

maybe it would be easier, at startup, to use `SIOCSIFFLAGS` to get the flags for every ifindex specified and then know for sure that a particular device is a loopback ?

Reply:
thank you, yes the intent was to not fetch napi ids for loopback as that would hit the `h2o_fatal` a few lines down when the `napi id == 0`, ill refactor

Reply:
changed to comparing against `lo` in ce30483e827c7ca770d88c5487b62604f17ae4dd.  `SIOCSIFFLAGS` would need a socket to run ioctl on